### PR TITLE
rewrote footer tests using Enzyme

### PR DESCRIPTION
**Pull request by tsl143**
rewrote footer tests using Enzyme

### CODE REVIEW

**Comment by willdurand** [comment]
@tsl143 are you familiar with `shallowUntilTarget` and Enzyme `shallow`? I think we dont need `mount` here and `shallowUntilTarget` should be enough.

Example: https://github.com/mozilla/addons-frontend/blob/07609f3a1255942ee4a24d42ed171b5b213d4b84/tests/unit/disco/components/TestFooter.js

You can still pass `store` to the component if needed.

**Comment by tsl143** [comment]
@willdurand thanks for reminding me, I used this before. We don't need to pass store though.
Shallow didn't help, but yes `shallowUntilTarget` is useful

File: tests/unit/amo/components/TestFooter.js [hunk 1]
```diff
@@ -32,19 +19,18 @@ describe('Footer', () => {
...
+    expect(root.find('.Footer-cookies-link').text()).toEqual('Cookies');
+    expect(root.find('.Footer-cookies-link').prop('href')).toEqual('https://www.mozilla.org/privacy/websites/#cookies');
+    expect(root.find('.Footer-trademark-abuse-link').text()).toEqual('Report Trademark Abuse');
+    expect(root.find('.Footer-trademark-abuse-link').prop('href')).toEqual('https://www.mozilla.org/about/legal/fraud-report/');
 
     // This link isn't localized because MDN will 404 on some
     // locales and not others.
     // See also https://bugzilla.mozilla.org/show_bug.cgi?id=1283422
-    expect(root.querySelector('.Footer-bug-report-link').href).toEqual('https://developer.mozilla.org/Add-ons/AMO/Policy/Contact');
+    expect(root.find('.Footer-bug-report-link').prop('href')).toEqual('https://developer.mozilla.org/Add-ons/AMO/Policy/Contact');
```
Comment (willdurand) [inline tests/unit/amo/components/TestFooter.js]: You could use `toHaveProp` thanks to [jest-enzyme](https://github.com/blainekasten/enzyme-matchers/blob/master/packages/jest-enzyme/README.md).

```
expect(root.find('.Footer-bug-report-link')).toHaveProp('href', 'https://developer.mozilla.org/Add-ons/AMO/Policy/Contact');
```

File: tests/unit/amo/components/TestFooter.js [hunk 2]
```diff
@@ -32,19 +19,18 @@ describe('Footer', () => {
...
-    expect(root.querySelector('.Footer-privacy-link').textContent).toEqual('Privacy');
-    expect(root.querySelector('.Footer-privacy-link').href).toEqual('https://www.mozilla.org/privacy/websites/');
-    expect(root.querySelector('.Footer-legal-link').textContent).toEqual('Legal');
-    expect(root.querySelector('.Footer-legal-link').href).toEqual('https://www.mozilla.org/about/legal/');
-    expect(root.querySelector('.Footer-cookies-link').textContent).toEqual('Cookies');
-    expect(root.querySelector('.Footer-cookies-link').href).toEqual('https://www.mozilla.org/privacy/websites/#cookies');
-    expect(root.querySelector('.Footer-trademark-abuse-link').textContent).toEqual('Report Trademark Abuse');
-    expect(root.querySelector('.Footer-trademark-abuse-link').href).toEqual('https://www.mozilla.org/about/legal/fraud-report/');
-
+    expect(root.find('.Footer-privacy-link').text()).toEqual('Privacy');
```
Comment (tofumatt) [inline tests/unit/amo/components/TestFooter.js]: This works but there's an even nicer helper you can use to write tests like this one:
```js
expect(root.find('.Footer-privacy-link')).toHaveText('Privacy');
```

File: tests/unit/amo/components/TestFooter.js [hunk 3]
```diff
@@ -32,19 +19,18 @@ describe('Footer', () => {
...
+    expect(root.find('.Footer-privacy-link').prop('href')).toEqual('https://www.mozilla.org/privacy/websites/');
```
Comment (tofumatt) [inline tests/unit/amo/components/TestFooter.js]: Same here, you should be able to use:
```js
expect(root.find('.Footer-privacy-link')).toHaveProp('href', 'https://www.mozilla.org/privacy/websites/');
```

**Comment by tsl143** [comment]
@tofumatt thanks a lot, these look much better :)

**Pull request merged**